File: max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        if(nums.size() == 1)return nums[0];
        int ans = 0;
        int temp = 0;
        for(int i = 0 ; i < nums.size() - 1 ; i++){
            if(nums[i] == 1 && nums[i+1] == 1){
                temp++;
            }else if(nums[i] == 1 && nums[i+1] == 0){
                ans = max(ans,temp+1);
                temp = 0;
            }else{
                ans = max(ans,temp);
                temp = 0;
            }
        }
        if(nums[nums.size() - 1] == 1){
            ans = max(ans,temp+1);
        }
        return ans;
    }
};

File: max-consecutive-ones/max-consecutive-ones.go
func max(i,j int)int{
    if i < j {
        return j;
    }
    return i;
}
func findMaxConsecutiveOnes(nums []int) int {
    curr_max,ans := 0,0
    length := len(nums)
    for i := 0 ; i < length ; i++ {
        if nums[i] == 1{
            curr_max++;
        }else{
            ans = max(ans,curr_max);
            curr_max = 0
        }
    } 
    ans = max(ans,curr_max);
    return ans;
}